src/test/java/org/fogbowcloud/manager/core/processors/SpawningProcessorTest.java

```diff
@@ -93,212 +92,291 @@ public void tearDown() {
+        ComputeInstance computeOrderInstance = Mockito.spy(new ComputeInstance(FAKE_INSTANCE_ID));
+        computeOrderInstance.setState(InstanceState.INACTIVE);
+        order.setInstanceId(FAKE_INSTANCE_ID);
 
-        Order test = this.closedOrderList.getNext();
-        Assert.assertEquals(OrderState.CLOSED, test.getOrderState());
-        Assert.assertNull(this.spawningOrderList.getNext());
-        Assert.assertNull(this.failedOrderList.getNext());
         Assert.assertNull(this.fulfilledOrderList.getNext());
+        
+        this.thread = new Thread(this.spawningProcessor);
+        this.thread.start();
+        Thread.sleep(500);
+
+        Order test = this.spawningOrderList.getNext();
+        Assert.assertNotNull(test);
+        Assert.assertEquals(order.getInstanceId(), test.getInstanceId());
+        Assert.assertEquals(OrderState.SPAWNING, test.getOrderState());
     }
 
     @Test
-    public void testRunConfiguredToFailedAttemptSshConnectivity() throws InterruptedException {
-        Order order = this.createOrder();
+    public void testRunProcessComputeOrderWhenInstanceStateIsFailed() throws Exception {
+        Order order = spyComputeOrder();
         order.setOrderState(OrderState.SPAWNING);
         this.spawningOrderList.addItem(order);
+        Mockito.doReturn(false).when(order).isRequesterRemote(Mockito.anyString());
+        
+        Instance orderInstance = Mockito.spy(new ComputeInstance(FAKE_INSTANCE_ID));
+        orderInstance.setState(InstanceState.FAILED);
+        order.setInstanceId(FAKE_INSTANCE_ID);
+        
+        Mockito.doReturn(orderInstance).when(this.cloudConnector).getInstance(
+                Mockito.any(Order.class));
 
-        String instanceId = "fake-id";
-        ComputeInstance computeOrderInstance =
-                spy(new ComputeInstance(instanceId));
-        computeOrderInstance.setState(InstanceState.READY);
-        order.setInstanceId(instanceId);
+        Mockito.when(this.tunnelingService.getExternalServiceAddresses(
+                Mockito.eq(order.getId()))).thenReturn(new HashMap<>());
 
-        Mockito.when(this.tunnelingService.getExternalServiceAddresses(Mockito.eq(order.getId())))
-				.thenReturn(new HashMap<>());
+        Assert.assertNull(this.failedOrderList.getNext());
+        
+        this.thread = new Thread(this.spawningProcessor);
+        this.thread.start();
+        Thread.sleep(500);
 
-        Mockito.when(this.sshConnectivity.checkSSHConnectivity(Mockito.any(
-        		SshTunnelConnectionData.class))).thenReturn(true);
+        Assert.assertNull(this.spawningOrderList.getNext());
+
+        Order test = this.failedOrderList.getNext();
+        Assert.assertNotNull(test);
+        Assert.assertEquals(order.getInstanceId(), test.getInstanceId());
+        Assert.assertEquals(OrderState.FAILED, test.getOrderState());
+    }
 
+    @Test
+    public void testRunProcessComputeOrderWhenOrderStateIsNotSpawning() throws Exception {
```

Comment:
What did you expect after the exercise? I can't understand what is being tested here.

Reply:
Test removed.